File: s08_Maps/MyApplication/app/src/main/java/com/omeryildizce/maps/roomdb/PlaceDao.java
package com.omeryildizce.maps.roomdb;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import com.omeryildizce.maps.model.Place;

import java.util.List;

import io.reactivex.rxjava3.core.Completable;
import io.reactivex.rxjava3.core.Flowable;

@Dao
public interface PlaceDao {
    @Query("Select * from Place")
    Flowable<List<Place>> getAll();

    @Insert
    Completable insert(Place place);

    @Delete
    Completable delete(Place place);


}

File: s21_Maps/KotlinMaps/app/src/main/java/com/omeryildizce/kotlinmaps/roomdb/PlaceDao.kt
package com.omeryildizce.kotlinmaps.roomdb

import androidx.room.*
import com.omeryildizce.kotlinmaps.model.Place
import io.reactivex.rxjava3.core.Completable
import io.reactivex.rxjava3.core.Flowable

@Dao
interface PlaceDao {
    @Query("select * from Place")
    fun getAll() : Flowable<List<Place>>

    @Insert
    fun insert(place: Place) : Completable

    @Update
    fun update(place: Place) : Completable

    @Delete
    fun delete(place: Place) :Completable


}